libs/verifiable-credentials/src/main/java/uk/gov/di/ipv/core/library/verifiablecredential/service/SessionCredentialsService.java

```diff
@@ -0,0 +1,83 @@
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.config.ConfigurationVariable;
+import uk.gov.di.ipv.core.library.config.EnvironmentVariable;
+import uk.gov.di.ipv.core.library.domain.ErrorResponse;
+import uk.gov.di.ipv.core.library.domain.VerifiableCredential;
+import uk.gov.di.ipv.core.library.exceptions.VerifiableCredentialException;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+import uk.gov.di.ipv.core.library.persistence.DataStore;
+import uk.gov.di.ipv.core.library.persistence.item.SessionCredentialItem;
+import uk.gov.di.ipv.core.library.service.ConfigService;
+
+import java.util.List;
+
+import static uk.gov.di.ipv.core.library.config.CoreFeatureFlag.SESSION_CREDENTIALS_TABLE_WRITES;
+
+public class SessionCredentialsService {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final DataStore<SessionCredentialItem> dataStore;
+    private final ConfigService configService;
+
+    public SessionCredentialsService(
+            DataStore<SessionCredentialItem> dataStore, ConfigService configService) {
+        this.dataStore = dataStore;
+        this.configService = configService;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public SessionCredentialsService(ConfigService configService) {
+        this.configService = configService;
+        this.dataStore =
+                new DataStore<>(
+                        configService.getEnvironmentVariable(
+                                EnvironmentVariable.SESSION_CREDENTIALS_TABLE_NAME),
+                        SessionCredentialItem.class,
+                        DataStore.getClient(),
+                        configService);
+    }
+
+    public void persistCredential(
+            VerifiableCredential credential, String ipvSessionId, boolean receivedThisSession)
+            throws VerifiableCredentialException {
+        try {
+            if (configService.enabled(SESSION_CREDENTIALS_TABLE_WRITES)) {
+                dataStore.create(
+                        credential.toSessionCredentialItem(ipvSessionId, receivedThisSession),
+                        ConfigurationVariable.SESSION_CREDENTIALS_TTL);
+            } else {
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                "Not writing to session credentials table as feature flag not active"));
+            }
+        } catch (Exception e) {
+            LOGGER.error("Error persisting session credential: {}", e.getMessage(), e);
+            throw new VerifiableCredentialException(
+                    HTTPResponse.SC_SERVER_ERROR, ErrorResponse.FAILED_TO_SAVE_CREDENTIAL);
+        }
+    }
+
+    public void persistCredentials(List<VerifiableCredential> vcs, String ipvSessionId)
+            throws VerifiableCredentialException {
+        for (var vc : vcs) {
+            persistCredential(vc, ipvSessionId, false);
```

Comment:
It feels a bit surprising that this sets `receivedThisSession = false` but the singular method requires you to specify.

Can we add the flag here as well, or create distinct methods for each type (e.g. `persistNewCredential` vs. `persistExistingCredential`)?

Reply:
Yep, this was mostly for convenience (when would we be storing multiple VC received this session?), but you're right it's probably better to be explicit.